credentials/generate-revocation-set.py

```diff
@@ -0,0 +1,243 @@
+
+        if is_paa == True:
+            if crl_vid != None:
+                if vid != crl_vid:
+                    continue
+        else:
+            if vid == None or vid != crl_vid:
+                continue
+            if crl_pid != None:
+                if pid != crl_pid:
+                    continue
+
+        # 5.
+        crl_signer_issuer_name = base64.b64encode(crl_signer_certificate.issuer.public_bytes()).decode('utf-8')
+
+        crl_signer_authority_key_id = crl_signer_certificate.extensions.get_extension_for_oid(
+            x509.OID_AUTHORITY_KEY_IDENTIFIER).value.key_identifier
+
+        paa_certificate = None
+
+        if use_rest:
+            response = requests.get(
+                f"{rest_node_url}/dcl/pki/certificates/{crl_signer_issuer_name}/{crl_signer_authority_key_id}").json()["approvedCertificates"]["certs"][0]
+            paa_certificate = response["pemCert"]
+        else:
+            cmdlist = ['query', 'pki', 'x509-cert', '-u', crl_signer_issuer_name, '-k', crl_signer_authority_key_id]
+            cmdpipe = subprocess.Popen(use_dcld(dcld, production, cmdlist), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            paa_certificate = json.loads(cmdpipe.stdout.read())["approvedCertificates"]["certs"][0]["pemCert"]
+
+        if paa_certificate == None:
+            continue
+
+        paa_certificate_object = x509.load_pem_x509_certificate(paa_certificate)
+
+        try:
+            crl_signer_certificate.verify_directly_issued_by(paa_certificate_object)
+        except Exception:
+            continue
+
+        # 6. Obtain the CRL
+        r = requests.get(revocation_point["dataURL"])
+        crl_file = x509.load_der_x509_crl(r.content)
+
+        # 7. Perform CRL File Validation
+        crl_authority_key_id = crl_file.extensions.get_extension_for_oid(x509.OID_AUTHORITY_KEY_IDENTIFIER).value.key_identifier
+        crl_signer_subject_key_id = crl_signer_certificate.extensions.get_extension_for_oid(
+            x509.OID_SUBJECT_KEY_IDENTIFIER).value.key_identifier
+        if crl_authority_key_id != crl_signer_subject_key_id:
+            continue
+
+        issuer_subject_key_id = ''.join('{:02X}'.format(x) for x in crl_authority_key_id)
+
+        same_issuer_points = None
+
+        if use_rest:
+            response = requests.get(
+                f"{rest_node_url}/dcl/pki/revocation-points/{issuer_subject_key_id}").json()["pkiRevocationDistributionPointsByIssuerSubjectKeyID"]
+            same_issuer_points = response["points"]
+        else:
+            cmdlist = ['query', 'pki', 'revocation-points', '--issuer-subject-key-id', issuer_subject_key_id]
+            cmdpipe = subprocess.Popen(use_dcld(dcld, production, cmdlist), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            same_issuer_points = json.loads(cmdpipe.stdout.read())[
+                "pkiRevocationDistributionPointsByIssuerSubjectKeyID"]["points"]
```

Comment:
Extract to a helper function. Can be done in a follow-up.

Reply:
Added comments in the src file to capture the follow-up work.